### PR TITLE
Feature/KAS-4167 Make the signature pills poll their status regularly

### PR DESCRIPTION
**Pull request by sergiofenoll**
Uses an equivalent construction to what we do in the publication pills to poll the status.

We might want to replace usage of models and relationships (`signMarkingActivity.signSubcase; signSubcase.signFlow; signFlow.status`) with just a query. The query won't be cached, but given that we're explicitly trying to reload data that's probably a good thing in this case (and it means we don't need to do a bunch of manual reloads to make the code work when ran a second time with the same `signMarkingActivity`).
TBD if we want to change this, now, some other time, or never.

### CODE REVIEW

File: app/components/signature-pill.js [hunk 1]
```diff
@@ -12,8 +17,30 @@ export default class SignaturePillComponent extends Component {
...
 
+  scheduledRefresh;
+  @tracked triggerTask;
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    cancel(this.scheduledRefresh);
+  }
+
+  schedulePublicationActivitiesRefresh() {
```
Comment (tdn) [inline app/components/signature-pill.js]: Perhaps better to rename this to `scheduleSignflowStatusRefresh`

File: app/components/signature-pill.js [hunk 2]
```diff
@@ -42,13 +69,15 @@ export default class SignaturePillComponent extends Component {
...
+    this.schedulePublicationActivitiesRefresh();
+
     return {
       signingHubUrl,
       status,
     };
   });
 
-  data = trackedTask(this, this.loadData);
+  data = trackedTask(this, this.loadData, () => [this.triggerTask]); // Make the resource dependant on this.triggerTask
```
Comment (tdn) [inline app/components/signature-pill.js:77]: I'm not sure I understand what's happening here.

So on line 72, on the first run of `this.loadData`, we load the first `status` and schedule the polling task. 
This will reload the `signFlow.status` after `SIGN_FLOW_STATUS_REFRESH_INTERVAL_MS` and set `this.triggerTask = new Date();`, which will trigger `this.loadData` because this trackedTask uses `this.triggerTask` as an argument ( https://ember-resources.pages.dev/funcs/util_ember_concurrency.trackedTask )

On line 43 in `this.loadData`, we then cancel the next `this.scheduledRefresh` to load the `status`, then we're back at line 72 and schedule the next task again.

So in other words, wouldn't it make more sense to put the reload code from lines 32 to 34 in `this.loadData` on lines 44 to 46? That way the scheduling task just updates the trigger for the trackedTask.
Or does this cause problems in Ember because you can't `reload()` on the first `await` of a relation?

**Review by tdn — CHANGES_REQUESTED** [review]
Just a small rename of the copied scheduling task, and just a look at my comment on how the scheduled task reloads data (perhaps it can be done in loadData?).

Ran the branch locally, and the update call seems to work well as far as I can test using mock login

File: app/components/signature-pill.js (hunk 2 above)
Comment (sergiofenoll) [reply app/components/signature-pill.js:77]: That also definitely works, I didn't want to change the original code too much. At that point we might be better off writing something like

**Review by tdn — APPROVED** [review]

**Review by ValenberghsSven — APPROVED** [review]
